pkg/registry/resource/resourceclaim/strategy.go

```diff
@@ -172,4 +174,49 @@ func toSelectableFields(claim *resource.ResourceClaim) fields.Set {
 
 // dropDisabledFields removes fields which are covered by a feature gate.
 func dropDisabledFields(newClaim, oldClaim *resource.ResourceClaim) {
+	dropDisabledDRAAdminAccessFields(newClaim, oldClaim)
+}
+
+func dropDisabledDRAAdminAccessFields(newClaim, oldClaim *resource.ResourceClaim) {
+	if utilfeature.DefaultFeatureGate.Enabled(features.DRAAdminAccess) {
+		// No need to drop anything.
+		return
+	}
+
+	for i := range newClaim.Spec.Devices.Requests {
```

Comment:
FWIW, I think this could just be:

```
if utilfeature.DefaultFeatureGate.Enabled(features.DRAAdminAccess) {
    return
}
if draAdminAccessFeatureInUse(oldClaim) {
    return
}

for i := range newClaim.Spec.Devices.Requests {
    newClaim.Spec.Devices.Requests[i].AdminAccess = nil
}
if newClaim.Status.Allocation == nil {
    return
}
for i := range newClaim.Status.Allocation.Devices.Results {
    newClaim.Status.Allocation.Devices.Results[i].AdminAccess = nil
}
```

Couldn't it?  :)

Reply:
Yes. My thinking was that if there is nothing to drop, then I don't need to call `draAdminAccessFeatureInUse`, which should be the most common case. But I am iterating over both slices either way, so that doesn't really make sense.

But it's good that this is merged because it unblocks next steps. I'll fix this in a [follow-up PR](https://github.com/kubernetes/kubernetes/pull/128433).